shell/platform/linux/public/flutter_linux/fl_dart_project.h

```diff
@@ -96,6 +96,34 @@ const gchar* fl_dart_project_get_assets_path(FlDartProject* project);
  */
 const gchar* fl_dart_project_get_icu_data_path(FlDartProject* project);
 
+/**
+ * fl_dart_project_set_dart_entrypoint_arguments:
+ * @project: an #FlDartProject.
+ * @argc: the number of command line arguments in @argv
+ * @argv: a pointer to an array of C strings containing the command line arguments.
+ *
+ * Sets the command line arguments to be passed through to the Dart
+ * entrypoint function.
+ *
+ * FlDartProject will not keep a deep copy nor ownership of these arguments;
+ * the caller is responsible for ensuring that they are not freed before the
+ * Flutter Engine is initialized, at which point they will be copied and can
+ * be safely freed.
```

Comment:
This smells strongly of premature optimization; copying should cost us essentially nothing in the overall cost of startup, and avoids the need for warnings about this API having non-standard semantics.

Reply:
I figured the use case to optimise for was to just pass through argc/argv unmodified from `int main(int argc, char** argv)` and the caller could just forget about it as they'd be good for the lifetime of the executable. I was on the fence either way, it just seemed easier to not bother with a copy.

I agree though, I'll add the copy anyway as it won't affect that usecase but makes things a little clearer.